platforms/osx/build_framework.py

```diff
@@ -26,11 +28,21 @@ def getBuildCommand(self, arch, target):
             "xcodebuild",
             "MACOSX_DEPLOYMENT_TARGET=" + os.environ['MACOSX_DEPLOYMENT_TARGET'],
             "ARCHS=%s" % arch,
-            "-sdk", target.lower(),
+            "-sdk", "macosx" if target == "Catalyst" else target.lower(),
             "-configuration", "Debug" if self.debug else "Release",
             "-parallelizeTargets",
             "-jobs", str(multiprocessing.cpu_count())
         ]
+
+        if target == "Catalyst":
+            buildcmd.append("-destination 'platform=macOS,arch=%s,variant=Mac Catalyst'" % arch)
+            buildcmd.append("-UseModernBuildSystem=YES")
+            buildcmd.append("SKIP_INSTALL=NO")
+            buildcmd.append("BUILD_LIBRARY_FOR_DISTRIBUTION=YES")
+            buildcmd.append("TARGETED_DEVICE_FAMILY=\"1,2\"")
+            buildcmd.append("SDKROOT=iphoneos")
+            buildcmd.append("SUPPORTS_MAC_CATALYST=YES")
+
```

Comment:
Depending on the results of https://gitlab.kitware.com/cmake/cmake/-/issues/21436 this whole block might not be necessary. If it is, we could still probably trim this down to a more minimal set of overrides because I think some of these are already handled elsewhere in the toolchain.